bots/sg_modify_scope_by_port.py

```diff
@@ -0,0 +1,189 @@
+"""
+##sg_modify_scope_by_port
+# What it does: modify Security Group's rules scope by a given port , new and old scope(optional).
+# Direction can be : inbound or outbound
+# Usage: AUTO: sg_modify_scope_by_port <port> <change_scope_from|*> <change_scope_to> <direction>
+       - When '*' set for replacing any rule with the specific port
+# Examples:
+        AUTO: sg_modify_scope_by_port 22 0.0.0.0/0 10.0.0.0/24 inbound
+        AUTO: sg_modify_scope_by_port 22 * 10.0.0.0/24 inbound
+#Notes:
+    -  if the port is in a rule's port range, the bot will change the rule's ip to desire ip , to avoid that
+      specify existing rule's scope instead of using '*'
+    - to split the rule around the port you can use the bot : #sg_single_rule_delete
+
+#Limitations: IPv6 is not supported yet
+
+"""
+import re
+
+PORT_TO = 'portTo'
+PORT_FROM = 'port'
+PROTOCOL = 'protocol'
+SCOPE = 'scope'
+ALL_TRAFFIC_PORT = 0
+ALL_TRAFFIC_PROTOCOL = '-1'
+
+"""
+checks if a rule exists in a security group , returns false/true 
+"""
+
+
+def is_rule_exists_in_sg(sg, rule, direction, scope):
+    if direction.lower() == 'inbound':
+        for perm in sg.ip_permissions:
```

Comment:
You have here the same code twice, move to function, let me know you're done and everything is tested so ill merge

Reply:
moved duplicate to a func